suite-native/module-device-settings/src/screens/FirmwareUpdateScreen/FirmwareChangelog.tsx

```diff
@@ -0,0 +1,80 @@
+import { useMemo } from 'react';
+import { useSelector } from 'react-redux';
+
+import { BottomSheet, Button, Text } from '@suite-native/atoms';
+import { selectFirmwareChangelog } from '@suite-common/wallet-core';
+import { Translation } from '@suite-native/intl';
+import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
+
+type FirmwareChangelogProps = {
+    isVisible: boolean;
+    onClose: () => void;
+};
+
+const changelogSectionTitleTextStyle = prepareNativeStyle(utils => ({
+    ...utils.typography.highlight,
+    paddingTop: utils.spacings.sp24,
+}));
+
+const buttonContainerStyle = prepareNativeStyle(utils => ({
+    marginTop: utils.spacings.sp32,
+}));
+
+const ChangelogSectionTitle = ({ children }: { children: React.ReactNode }) => {
+    const { applyStyle } = useNativeStyles();
+
+    return <Text style={applyStyle(changelogSectionTitleTextStyle)}>{children}</Text>;
+};
+
+export const FirmwareChangelog = ({ isVisible, onClose }: FirmwareChangelogProps) => {
+    const firmwareChangelog = useSelector(selectFirmwareChangelog);
+    const { applyStyle } = useNativeStyles();
+
+    const formattedChangelog = useMemo(() => {
+        if (!firmwareChangelog) {
+            return (
+                <Text>
+                    <Translation id="moduleDeviceSettings.firmware.firmwareUpdateScreen.changelog.changelogUnavailable" />
+                </Text>
+            );
+        }
+
+        if (typeof firmwareChangelog === 'string') {
+            return <Text>{firmwareChangelog}</Text>;
+        }
+
+        return firmwareChangelog.map((text, index) => {
+            const key = text + index;
+
+            if (text.startsWith('#')) {
+                const strippedText = text.replaceAll('#', '').trim();
+
+                return <ChangelogSectionTitle key={key}>{strippedText}</ChangelogSectionTitle>;
+            }
+
+            let formattedText = text;
+
+            if (text.startsWith(' - ')) {
+                formattedText = formattedText.replace(' - ', ' • ');
+            }
+
+            return <Text key={key}>{formattedText}</Text>;
+        });
```

Comment:
This seems to be a great candidate for a utility function.

Reply:
I don't think we will reuse this code anywhere else and component itself is basically only this function, so I think there is no reason create extra function for it.